### PR TITLE
Feature/infinite scroll in reports

### PR DESCRIPTION
**Pull request by tanmaythole**
create pagination on the reports page

**Summary**
When the user navigated to the reports page to view the previous reports, all the reports were fetched from the backend and shown to the user. This PR paginates the report page when the user scrolls down, and the new API request call and fetch the data for the next page.

Ticket Link:
https://trello.com/c/qWFeXfY3/20-create-pagination

**Solution:**
This is done using _react-infinite-scroll-component_ library.

### CODE REVIEW

File: src/Components/CustomComponents/InfiniteScrollComponent.jsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+                </div>
+            )}
+            {...props}
+        >
+            {children}
+        </InfiniteScroll>
+    );
+};
+
+export default InfiniteScrollComponent;
```
Comment (ChinmaySonawane) [inline src/Components/CustomComponents/InfiniteScrollComponent.jsx]: export should be at the end of the file

Comment (tanmaythole) [reply src/Components/CustomComponents/InfiniteScrollComponent.jsx]: Done

File: package.json [hunk 2]
```diff
@@ -14,6 +14,7 @@
     "react-bootstrap": "^2.4.0",
     "react-bootstrap-icons": "^1.8.2",
     "react-dom": "^18.1.0",
+    "react-infinite-scroll-component": "^6.1.0",
```
Comment (ChinmaySonawane) [inline package.json]: why we are using this package?
we can also implement this infinite-scroll.
Lets discuss this...

File: src/Components/CustomComponents/InfiniteScrollComponent.jsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import InfiniteScroll from 'react-infinite-scroll-component';
+import PropTypes from 'prop-types';
+import React from 'react';
+import Spinner from 'react-bootstrap/Spinner';
+
+const InfiniteScrollComponent = ({ children, loader, ...props }) => {
+    return (
```
Comment (ChinmaySonawane) [inline src/Components/CustomComponents/InfiniteScrollComponent.jsx]: As we don't need any destructuring here we can use arrow function with a single line return.
i.e const functionName = () => (returningValue);

File: package.json (hunk 2 above)
Comment (tanmaythole) [reply package.json]: custom infinite scroll added using Intersection observer API

File: src/Components/CustomComponents/InfiniteScrollComponent.jsx (hunk 3 above)
Comment (tanmaythole) [reply src/Components/CustomComponents/InfiniteScrollComponent.jsx]: done 👍🏻

File: src/store/Reports/reducers.js [hunk 4]
```diff
@@ -1,101 +1,75 @@
-import reportsConstants from "./constants";
+import reportsConstants from './constants';
 
 const INITIAL_STATE = {
     isReportsLoading: true,
     reports: [],
+    totalResults: 0,
+    page: 0,
+    pageSize: 20,
```
Comment (ChinmaySonawane) [inline src/store/Reports/reducers.js]: can we use PAGE_SIZE constant here instead of value?

Comment (tanmaythole) [reply src/store/Reports/reducers.js]: pageSize in reducers is unused, so removed

**Review by ChinmaySonawane — APPROVED** [review]

**Pull request merged**